### PR TITLE
Feature/mvp 12 search atf

### PR DESCRIPTION
**Pull request by danieliordache**
update angular and ionic
update dependencies
refactor css
refactor project structure and providers
implement atf-search functionality 
implement unit/integration/e2e test config
implement unit tests

### CODE REVIEW

File: src/directives/transforming-searchbar/transforming-searchbar.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  ngOnDestroy() {
+    this.searchBarElemRef.ionFocus.unsubscribe();
+    this.searchBarElemRef.ionCancel.unsubscribe();
+    this.events.unsubscribe('navToDetails');
+  }
+
+  private setDefaultCss(scrollContent, navBarElement): void {
+    this.renderer.setStyle(scrollContent, 'margin-top', '130px');
```
Comment (acozianu) [inline src/directives/transforming-searchbar/transforming-searchbar.ts]: Use css class toggle for adding style to an element.

File: src/directives/transforming-searchbar/transforming-searchbar.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+export class TransformingSearchBarDirective implements AfterViewInit, OnDestroy {
+  @Input() searchBarElemRef
+  headerElemRef;
+
+  constructor(private el: ElementRef, private renderer: Renderer2, public events: Events) {
+    this.headerElemRef = el.nativeElement;
+  }
+
+  ngAfterViewInit() {
+    let navBarElement = this.headerElemRef.querySelector('ion-navbar');
```
Comment (acozianu) [inline src/directives/transforming-searchbar/transforming-searchbar.ts]: If the value doesn't change, you should use consts.

File: src/pages/testing/test-creation/complete-test/complete-test.scss [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
+    padding: 0;
   }
 
   .scroll-content {
     max-height: calc(100% - 215px);
   }
 
   ion-list-header {
-    background-color: whitesmoke;
     font-size: 1.3rem;
```
Comment (acozianu) [inline src/pages/testing/test-creation/complete-test/complete-test.scss]: we should use the same metric system in the entire project.
You should choose one of them and do the refactor.
From my point of view, px should be ok for the project complexity

File: src/providers/search.service.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+              (elem: AtfModel) => {
+                elem.searchProperty = properties[0];
+              }
+            )
+          }
+        )
+      }
+      return this.orderedArray.length > 0 ? this.orderedArray : this.groupByLetter(items, properties[0]);
+    }
+    for (let i = 0; i < items.length; i++) {
```
Comment (acozianu) [inline src/providers/search.service.ts]: use forEach

File: src/directives/transforming-searchbar/transforming-searchbar.ts (hunk 2 above)
Comment (danieliordache) [reply src/directives/transforming-searchbar/transforming-searchbar.ts]: done

File: src/directives/transforming-searchbar/transforming-searchbar.ts (hunk 1 above)
Comment (danieliordache) [reply src/directives/transforming-searchbar/transforming-searchbar.ts]: I've tried to add the css classes but it didn't work.

File: src/pages/testing/test-creation/complete-test/complete-test.scss (hunk 3 above)
Comment (danieliordache) [reply src/pages/testing/test-creation/complete-test/complete-test.scss]: I've modified only in the files affected by the user story, the rest will be modified in their stories.

File: src/providers/search.service.ts (hunk 4 above)
Comment (danieliordache) [reply src/providers/search.service.ts]: done

File: src/app/app.module.ts [hunk 5]
```diff
@@ -27,30 +61,20 @@ import {HttpClientModule} from "@angular/common/http";
   imports: [
     BrowserModule,
     HttpClientModule,
-    IonicModule.forRoot(MyApp)
+    // IonicModule.forRoot(MyApp, {statusbarPadding: true}),
```
Comment (acozianu) [inline src/app/app.module.ts]: Please fix the padding problem and remove the comment

File: src/pages/atf/atf-home/atf-home.scss [hunk 6]
```diff
@@ -1,67 +1,29 @@
...
-        font-family: "Arial Black", Gadget, sans-serif;
-        line-height: 40px;
-      }
+  .start-title {
+    padding-top: 10px;
+    h1 {
+      font-size: 26px;
+      font-weight: bold;
+      margin: 0;
+      font-family: "Arial Black", Gadget, sans-serif;
```
Comment (acozianu) [inline src/pages/atf/atf-home/atf-home.scss]: Please use font variables

File: src/providers/global/auth.service.ts [hunk 7]
```diff
@@ -1,32 +1,32 @@
...
   authenticate(): Promise<any> {
     return new Promise((resolve, reject) => {
       let poolData = {
-        UserPoolId: this.restrictedConfig.auth.userPoolId,
-        ClientId: this.restrictedConfig.auth.clientId
+        UserPoolId: AUTH.USER_POOL_ID,
+        ClientId: AUTH.CLIENT_ID
       };
       let isp = new CognitoIdentityServiceProvider();
       isp.config.region = 'eu-west-1';
```
Comment (acozianu) [inline src/providers/global/auth.service.ts]: Add config region in the enum file.

File: src/app/app.module.ts (hunk 5 above)
Comment (danieliordache) [reply src/app/app.module.ts]: fixed

File: src/pages/atf/atf-home/atf-home.scss (hunk 6 above)
Comment (danieliordache) [reply src/pages/atf/atf-home/atf-home.scss]: fixed

File: src/providers/global/auth.service.ts (hunk 7 above)
Comment (danieliordache) [reply src/providers/global/auth.service.ts]: fixed

**Pull request merged**